src/error.rs

```diff
@@ -19,13 +19,18 @@ pub type Result<T, E = Error> = std::result::Result<T, E>;
 /// Internal error.
 #[derive(Debug, Error)]
 #[allow(missing_docs)]
+#[allow(clippy::large_enum_variant)]
 pub enum Error {
     #[error("Failed signature check.")]
     FailedSignature,
     #[error("Cannot route.")]
     CannotRoute,
-    #[error("Network layer error: {0}")]
-    Network(#[from] qp2p::Error),
+    #[error("The config is invalid")]
+    InvalidConfig,
+    #[error("Cannot connect to the endpoint")]
+    CannotConnectEndpoint,
```

Comment:
Perhaps we could add some context to these errors to help identify the cause using:
`.map_err(|err| ErrorVariant(err.to_string()))`

Reply:
yeh will do that as part of the next step as I need to do it to the other variants as well